src/nebula/Connection.ts

```diff
@@ -143,14 +143,17 @@ export default class Connection extends EventEmitter {
         const timer = setTimeout(() => { resolve(false) }, timeout)
         this
           .client
-          .execute(0, 'YIELD 1')
-          .then(() => {
+          .execute(this.sessionId, 'YIELD 1')
+          .then(response => {
+            if (response.error_code !== 0) {
+              throw new NebulaError(response.error_code, response.error_msg)
+            }
             clearTimeout(timer)
             resolve(true)
           })
-          .catch(() => {
+          .catch(err => {
             clearTimeout(timer)
-            resolve(false)
+            reject(err)
```

Comment:
I figure that any error, including ones on connection lost should propagate since it's technically an error, even though it's not a user query. What behavior do you think there should be?

Reply:
ping roles:
1. keep alive for sessionId
2. detect connection lost
 
if response.error_code !== 0, we should reprepare connection.
if ping failed by connection lost, thrift.connection's error event will be trigger, maybe duplicated.

the ping is a internal mechanism for keep connection connective